tinygrad/codegen/kernel.py

```diff
@@ -470,6 +463,25 @@ def required_optimizations(self):
         self.apply_opt(Opt(OptOps.UPCAST, unit_stride_axes_mul_4[0], 4))
 
   def hand_coded_optimizations(self):
+    try: # check TC first and apply hand-coded opts if successful
```

Comment:
Can this just work like how it used to work? I don't see why this has to be in hand coded.